river/tree/_utils/feature_quantizer.py

```diff
@@ -0,0 +1,41 @@
+import math
+
+from .gradient_hessian import GradHess, GradHessStats
+
+
+class FeatureQuantizer:
```

Comment:
We need Henry to confirm this. What is the `observer` type in the paper?

Reply:
As far as I am concerned (please, correct me @henrygouk if I'm wrong), MOA's SGT uses a calibration set to define split points. This set has a user-defined size and consists of the first samples of the stream.

Reply:
Yes, in the paper the first 1000 examples of the stream are used to estimate the min/max values in order to do equal width binning.